modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustClientCodegen.java

```diff
@@ -399,43 +450,9 @@ public String modelDocFileFolder() {
 
     @Override
     public String getTypeDeclaration(Schema p) {
+        // use unaliased schema for client-side
         Schema unaliasSchema = unaliasSchema(p);
-        if (ModelUtils.isArraySchema(unaliasSchema)) {
-            ArraySchema ap = (ArraySchema) unaliasSchema;
-            Schema inner = ap.getItems();
-            if (inner == null) {
-                LOGGER.warn("{}(array property) does not have a proper inner type defined.Default to string",
-                        ap.getName());
-                inner = new StringSchema().description("TODO default missing array inner type to string");
-            }
-            return "Vec<" + getTypeDeclaration(inner) + ">";
-        } else if (ModelUtils.isMapSchema(unaliasSchema)) {
-            Schema inner = ModelUtils.getAdditionalProperties(unaliasSchema);
-            if (inner == null) {
-                LOGGER.warn("{}(map property) does not have a proper inner type defined. Default to string", unaliasSchema.getName());
-                inner = new StringSchema().description("TODO default missing map inner type to string");
-            }
-            return "::std::collections::HashMap<String, " + getTypeDeclaration(inner) + ">";
-        }
-
-        // Not using the supertype invocation, because we want to UpperCamelize
-        // the type.
-        String schemaType = getSchemaType(unaliasSchema);
-        if (typeMapping.containsKey(schemaType)) {
-            return typeMapping.get(schemaType);
-        }
-
-        if (typeMapping.containsValue(schemaType)) {
-            return schemaType;
-        }
-
-        if (languageSpecificPrimitives.contains(schemaType)) {
-            return schemaType;
-        }
-
-        // return fully-qualified model name
-        // crate::models::{{classnameFile}}::{{classname}}
-        return "crate::models::" + toModelName(schemaType);
```

Comment:
This returned the fully-qualified model name before, yet don't see any changes in samples/ perhaps this code path was uncovered by the existing sample?

Reply:
There are a lot of changes in sample related to imports, see my comment above:
https://github.com/OpenAPITools/openapi-generator/pull/17915#discussion_r1497538208

It's now aligned for Rust codegens

Reply:
It's compatible change, but unification of codegens made this less error-prone

Reply:
Particularly this codepath was mostly tested by `petstore`, for example in main, before this change: https://github.com/OpenAPITools/openapi-generator/blob/f1919042974ec17821d66a5529e4a8df5d3b3ac7/samples/client/petstore/rust/reqwest/petstore-async/src/models/pet.rs#L20